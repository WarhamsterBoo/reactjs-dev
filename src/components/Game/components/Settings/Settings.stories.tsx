import { action } from "@storybook/addon-actions";
import React from "react";
import { Settings } from "./Settings";

export default { title: "Settings Component" };

export const SettingsStory: React.FC<{}> = () => {
  return <Settings onSubmit={action("onSubmit")} />;
};
