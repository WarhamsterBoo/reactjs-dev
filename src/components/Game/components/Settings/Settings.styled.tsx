import styled from "@emotion/styled";

export const SettingsContainer = styled.div`
  width: 1300px;
  display: flex;
  flex-direction: row;
`;
