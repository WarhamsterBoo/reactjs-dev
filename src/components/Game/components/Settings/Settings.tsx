import React, { useCallback } from "react";
import { InputNumber } from ".";
import { SettingsContainer } from "./Settings.styled";

export interface SettingsProps {
  onSubmit: () => void;
}

export const Settings: React.FC<SettingsProps> = ({ onSubmit }) => {
  const onHandleSubmit = useCallback(
    (ev: React.FormEvent) => {
      ev.preventDefault();
      onSubmit();
    },
    [onSubmit]
  );

  return (
    <SettingsContainer>
      <form onSubmit={onHandleSubmit}>
        <fieldset>
          <legend>Game settings</legend>
          <label>
            X dimension:
            <InputNumber />
          </label>
          <label>
            Y dimension:
            <InputNumber />
          </label>
          <label>
            Filling Percentage:
            <InputNumber />
          </label>
          <button>Apply</button>
        </fieldset>
      </form>
    </SettingsContainer>
  );
};
