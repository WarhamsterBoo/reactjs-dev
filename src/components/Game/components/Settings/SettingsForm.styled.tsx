import styled from "@emotion/styled";

export const InputsContainer = styled.div`
  width: 1300px;
  display: flex;
  flex-direction: row;
`;
