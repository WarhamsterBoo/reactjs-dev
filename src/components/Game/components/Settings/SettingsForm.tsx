import React, { useCallback } from "react";
import { InputNumber } from ".";
import { InputsContainer } from "./SettingsForm.styled";

export interface SettingsFormProps {
  onSubmit: () => void;
}

export const SettingsForm: React.FC<SettingsFormProps> = ({ onSubmit }) => {
  const onHandleSubmit = useCallback(
    (ev: React.FormEvent) => {
      ev.preventDefault();
      onSubmit();
    },
    [onSubmit]
  );

  return (
    <InputsContainer>
      <form onSubmit={onHandleSubmit}>
        <fieldset>
          <legend>Game settings</legend>
          <label>
            X dimension:
            <InputNumber />
          </label>
          <label>
            Y dimension:
            <InputNumber />
          </label>
          <label>
            Filling Percentage:
            <InputNumber />
          </label>
          <button>Apply</button>
        </fieldset>
      </form>
    </InputsContainer>
  );
};
