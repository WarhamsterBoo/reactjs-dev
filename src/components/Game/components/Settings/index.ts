export * from "./SettingsForm";
export * from "./components";
