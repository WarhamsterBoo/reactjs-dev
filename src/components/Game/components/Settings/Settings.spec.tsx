import React from "react";
import renderer from "react-test-renderer";
import { Settings } from "./Settings";

describe("Settings", () => {
  it("should render", () => {
    const sut = <Settings onSubmit={() => {}} />;

    expect(renderer.create(sut).toJSON()).toMatchSnapshot();
  });
});
