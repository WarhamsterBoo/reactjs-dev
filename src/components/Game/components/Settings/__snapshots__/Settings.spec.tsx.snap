// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings should render 1`] = `
.emotion-3 {
  width: 1300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  border-radius: 4px;
  width: 50px;
  margin-bottom: 5px;
  margin-left: 5px;
  margin-right: 25px;
}

<div
  className="emotion-3"
>
  <form
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Game settings
      </legend>
      <label>
        X dimension:
        <input
          className="emotion-0"
          max="100"
          min="1"
          type="number"
        />
      </label>
      <label>
        Y dimension:
        <input
          className="emotion-0"
          max="100"
          min="1"
          type="number"
        />
      </label>
      <label>
        Filling Percentage:
        <input
          className="emotion-0"
          max="100"
          min="1"
          type="number"
        />
      </label>
      <button>
        Apply
      </button>
    </fieldset>
  </form>
</div>
`;
